From 65bd80861795471674c6a2eb50baf30d26576205 Mon Sep 17 00:00:00 2001
From: Artem Goncharov <artem.goncharov@gmail.com>
Date: Wed, 1 Jul 2026 17:07:15 +0200
Subject: [PATCH] Replace deprecated `timeutils.set_time_override`

`timeutils.set_time_override` is deprecated causing the tests failures.
Rewrites TimeFixture to mock timeutils.utcnow + timeutils.utcnow_ts
directly via mock.patch.object

Change-Id: I284683d654a0e5bce344cd094be71f5a1844732d
Signed-off-by: Artem Goncharov <artem.goncharov@gmail.com>
---
 .../auth_token/test_auth_token_middleware.py  | 21 +++++++++++++++----
 1 file changed, 17 insertions(+), 4 deletions(-)

diff --git a/keystonemiddleware/tests/unit/auth_token/test_auth_token_middleware.py b/keystonemiddleware/tests/unit/auth_token/test_auth_token_middleware.py
index 33e0912..6c20e7a 100644
--- a/keystonemiddleware/tests/unit/auth_token/test_auth_token_middleware.py
+++ b/keystonemiddleware/tests/unit/auth_token/test_auth_token_middleware.py
@@ -12,6 +12,7 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
+import calendar
 import datetime
 import os
 import time
@@ -136,8 +137,20 @@ class TimeFixture(fixtures.Fixture):
 
     def setUp(self):
         super(TimeFixture, self).setUp()
-        timeutils.set_time_override(self.new_time)
-        self.addCleanup(timeutils.clear_time_override)
+        patcher = mock.patch.object(timeutils, "utcnow", lambda: self.new_time)
+        patcher.start()
+        self.addCleanup(patcher.stop)
+        ts_patcher = mock.patch.object(
+            timeutils,
+            "utcnow_ts",
+            lambda microsecond=False: calendar.timegm(
+                self.new_time.timetuple()),
+        )
+        ts_patcher.start()
+        self.addCleanup(ts_patcher.stop)
+
+    def advance_time_seconds(self, seconds):
+        self.new_time += datetime.timedelta(seconds=seconds)
 
 
 class FakeApp(object):
@@ -606,11 +619,11 @@ class CommonAuthTokenMiddlewareTest(object):
         req.environ.update(extra_environ)
 
         now = datetime.datetime.now(datetime.timezone.utc)
-        self.useFixture(TimeFixture(now))
+        time_fixture = self.useFixture(TimeFixture(now))
         req.get_response(self.middleware)
         self.assertIsNotNone(self._get_cached_token(token))
 
-        timeutils.advance_time_seconds(token_cache_time)
+        time_fixture.advance_time_seconds(token_cache_time)
         self.assertIsNone(self._get_cached_token(token))
 
     def test_swift_memcache_set_expired(self):
-- 
2.55.0

